apps/test/unit/blockUtilsTest.js

```diff
@@ -853,5 +853,203 @@ describe('block utils', () => {
         }).to.throw(Error);
       });
     });
+    describe('methodCall', () => {
+      it('generates code for a method call', () => {
+        createBlock({
+          func: 'bark',
+          methodCall: true,
+          blockText: '{THIS} barks',
+          args: [],
+        });
+        const valueToCodeStub = sinon.stub(Blockly.JavaScript, 'valueToCode')
+          .callsFake((block, name) => {
+            return {
+              THIS: 'myDog',
+            }[name];
+          });
+        const code = generator['test_bark']();
+
+        expect(code.trim()).to.equal('myDog.bark();');
+
+        valueToCodeStub.restore();
+      });
+      it('generates code for a object property', () => {
+        createBlock({
+          name: 'getToy',
+          expression: 'favoriteToy',
+          methodCall: true,
+          blockText: '{THIS}\'s favorite toy',
+          args: [],
+          returnType: 'String',
+        });
+        const valueToCodeStub = sinon.stub(Blockly.JavaScript, 'valueToCode')
+          .callsFake((block, name) => {
+            return {
+              THIS: 'myDog',
+            }[name];
+          });
+        const [code,] = generator['test_getToy']();
+
+        expect(code.trim()).to.equal('myDog.favoriteToy');
+
+        valueToCodeStub.restore();
+      });
+    });
+    describe('eventBlock', () => {
+      it('generates code for an event block', () => {
+        createBlock({
+          func: 'whenJump',
+          blockText: 'when jump',
+          args: [],
+          eventBlock: true,
+        });
+        const blockToCodeStub = sinon.stub(Blockly.JavaScript, 'blockToCode')
+          .callsFake(() => {
+            return 'someHandlerCode();\n';
+          });
+        const code = generator['test_whenJump']();
+
+        expect(code.trim()).to.equal(dedent`
+          whenJump(function () {
+            someHandlerCode();
+          });`
+        );
+
+        blockToCodeStub.restore();
+      });
+    });
+    describe('expression blocks', () => {
+      it('generates code for an expression block', () => {
+        createBlock({
+          name: 'useStrict',
+          expression: 'use strict;',
+          blockText: 'run this program in strict mode',
+          args: [],
+        });
+        const code = generator['test_useStrict']();
+
+        expect(code.trim()).to.equal('use strict;');
+      });
+      it('generates code for an expression block with a return value', () => {
+        createBlock({
+          name: 'window',
+          expression: 'window',
+          blockText: 'window object',
+          args: [],
+          returnType: 'Object',
+        });
+        const [code,] = generator['test_window']();
```

Comment:
Is the comma in `[code,]` needed to only destructure the first value?

Reply:
Yup